robot/src/main/java/org/frcteam2910/common/robot/UpdateManager.java

```diff
@@ -0,0 +1,47 @@
+package org.frcteam2910.common.robot;
+
+import edu.wpi.first.wpilibj.Notifier;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public final class UpdateManager {
+	private final List<Updatable> updatables = new ArrayList<>();
+
+	@FunctionalInterface
+	private interface Updatable {
+		void update(double time, double dt);
+	}
+
+	private double lastTimestamp = 0.0;
+
+	private final Notifier kinematicThread = new Notifier(() -> {
```

Comment:
kinematics isn't really what this thread does anymore. Maybe this could be renamed to something like `updateNotifier`?